### ISSUE TITLE
Test `providers::aws::mock_tests::test_aws_imds_versions` is flaky

### ISSUE THREAD

**Issue by jlebon** [coreos/afterburn#714]
Saw it fail in a COPR build:

```
failures:

---- providers::aws::mock_tests::test_aws_imds_versions stdout ----
thread 'providers::aws::mock_tests::test_aws_imds_versions' panicked at 'called `Result::unwrap()` on an `Err` value: maximum number of retries (0) reached

Caused by:
    failed to fetch: 501 Not Implemented', src/providers/aws/mock_tests.rs:183:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    providers::aws::mock_tests::test_aws_imds_versions

```

Permalink to referenced line:

https://github.com/coreos/afterburn/blob/fd9e1ee761d9b4111479f99b9a305208c504c8f8/src/providers/aws/mock_tests.rs#L183

This may be related to the flake in https://github.com/coreos/afterburn/issues/705 but keeping it separate for now.

**Comment by lucab** [comment]
I'm almost running out of ideas, and starting to feel this may be either due to poor rng or to wrong optimization in release mode.

My next attempt would be splitting that test in two separate imds v1 and v2 ones, and doing an explicit `drop(mocks)` before the `reset()`.

**Comment by jlebon** [comment]
Let's give it a shot. Opened https://github.com/coreos/afterburn/pull/716.

**Comment by jlebon** [comment]
We haven't hit this again so far, so let's optimistically close as fixed by #716 for now.
